app/src/main/kotlin/edu/stanford/bdh/engagehf/health/components/NumberPicker.kt

```diff
@@ -0,0 +1,90 @@
+package edu.stanford.bdh.engagehf.health.components
+
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.width
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.foundation.lazy.itemsIndexed
+import androidx.compose.foundation.lazy.rememberLazyListState
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.derivedStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.rememberCoroutineScope
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.alpha
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.unit.dp
+import edu.stanford.spezi.core.design.theme.Colors
+import edu.stanford.spezi.core.design.theme.Spacings
+import edu.stanford.spezi.core.design.theme.SpeziTheme
+import edu.stanford.spezi.core.design.theme.TextStyles
+import edu.stanford.spezi.core.design.theme.ThemePreviews
+import kotlinx.coroutines.launch
+
+@Composable
+fun NumberPicker(
+    value: Int,
+    onValueChange: (Int) -> Unit,
+    range: IntRange,
+    modifier: Modifier = Modifier,
+) {
+    val coroutineScope = rememberCoroutineScope()
+    val lazyListState = rememberLazyListState(initialFirstVisibleItemIndex = value - 1)
+
+    Box(modifier = modifier, contentAlignment = Alignment.Center) {
+        LazyColumn(
+            state = lazyListState,
+            modifier = Modifier.fillMaxSize(),
+            horizontalAlignment = Alignment.CenterHorizontally,
+        ) {
+            itemsIndexed(range.toList()) { index, item ->
+                @Suppress("MagicNumber")
```

Comment:
Would extract a new property here: `val isFocusedItem = index == remember { derivedStateOf { lazyListState.firstVisibleItemIndex + 1 } }.value` and use below, on the text `color = Colors.onSurface.copy(alpha = if (isFocusedItem) 1f else 0.6f)`, same on modifier alpha